external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -2147,20 +2368,64 @@ fn parse_optional_type_parameters(
 //    StructTypeParameter = '<' Comma<TypeParameterWithPhantomDecl> ">" | <empty>
 fn parse_struct_type_parameters(
     context: &mut Context,
-) -> Result<Vec<StructTypeParameter>, Box<Diagnostic>> {
+) -> Result<Vec<DatatypeTypeParameter>, Box<Diagnostic>> {
     if context.tokens.peek() == Tok::Less {
         parse_comma_list(
             context,
             Tok::Less,
             Tok::Greater,
-            parse_type_parameter_with_phantom_decl,
+            parse_struct_type_parameter,
             "a type parameter",
         )
     } else {
         Ok(vec![])
     }
 }
 
+// Parse type parameter with optional phantom declaration:
+//   TypeParameterWithPhantomDecl = "phantom"? <TypeParameter>
+fn parse_struct_type_parameter(
+    context: &mut Context,
+) -> Result<DatatypeTypeParameter, Box<Diagnostic>> {
+    let (is_phantom, name, constraints) = parse_type_parameter_with_phantom_decl(context)?;
+    Ok(DatatypeTypeParameter {
+        is_phantom,
+        name,
+        constraints,
+    })
+}
+
+// Parse optional enum type parameters:
+//    EnumTypeParameter = '<' Comma<TypeParameterWithPhantomDecl> ">" | <empty>
+fn parse_enum_type_parameters(
+    context: &mut Context,
+) -> Result<Vec<DatatypeTypeParameter>, Box<Diagnostic>> {
+    if context.tokens.peek() == Tok::Less {
+        parse_comma_list(
+            context,
+            Tok::Less,
+            Tok::Greater,
+            parse_enum_type_parameter,
+            "a type parameter",
+        )
+    } else {
+        Ok(vec![])
+    }
+}
+
+// Parse type parameter with optional phantom declaration:
+//   TypeParameterWithPhantomDecl = "phantom"? <TypeParameter>
+fn parse_enum_type_parameter(
+    context: &mut Context,
+) -> Result<DatatypeTypeParameter, Box<Diagnostic>> {
+    let (is_phantom, name, constraints) = parse_type_parameter_with_phantom_decl(context)?;
+    Ok(DatatypeTypeParameter {
+        is_phantom,
+        name,
+        constraints,
+    })
+}
```

Comment:
This seems like an exact copy of `parse_struct_type_parameter`. Can this function just be removed and `parse_struct_type_parameter` be renamed to something like `parse_datatype_type_parameter`?

Same for `parse_enum_type_parameters`/`parse_struct_type_parameters`.